### ISSUE TITLE
CA1053 fires on types inheriting from a type with instance methods

### ISSUE THREAD

**Issue by hvanbakel** [dotnet/roslyn#4087]
CA1053 should only fire on types that do not derive from anything. The current code does not check inheritance. As a result a simple custom exception causes this rule to fire, if you only implement the constructors.

Consider the following code:
```
[Serializable]
public class TestException : Exception
{
    public TestException(string message)
        : base(message)
    {
    }

    public TestException(string message, Exception innerException)
        : base(message, innerException)
    {
    }

    protected TestException(SerializationInfo info, StreamingContext context)
        : base(info, context)
    {
    }
}
```

**Comment by hvanbakel** [comment]
I'm unable to reproduce this in a unit test. Seems like a false positive.
